Make Elements panel feel less cluttered

bvaughn opened the pull request:
### Do not merge before PR #1069

Resolves #1079

Includes the following subjective tweaks:
* Tweaks margins and paddings to feel less cluttered
* Re-styles header toolbar so search feels more prominent
* Moves "Highlight updates" and "Highlight search" options into settings popup
* Changes attribute display behavior to only show strings/numbers/booleans and hide the rest from the tree view
* Rather than forcing single-line elements, added improved wrapping behavior to make tree easier to see at a glance
* Forces breadcrumb strip to only take up a single vertical line rather than wrapping
* Improves scroll-into-view functionality to also horizontally scroll

### New search toolbar and settings
![settings](https://user-images.githubusercontent.com/29597/44056251-8c70122a-9efc-11e8-9494-dc14998207d3.gif)

### New element attribute wrapping behavior
![wrap](https://user-images.githubusercontent.com/29597/44056246-896aa392-9efc-11e8-92e4-2c0e868f7c49.gif)

### Improved breadcrumbs and scroll-to behavior
![breadcrumbs-and-snapping](https://user-images.githubusercontent.com/29597/44056241-85df80a8-9efc-11e8-9b93-7293c5df32a2.gif)

gaearon commented:
A few minor things based on screen recordings above:

* Styles appear wrong here (I mentioned on twitter but not sure if you have a chance to fix this yet)

<img width="532" alt="screen shot 2018-08-14 at 6 09 09 pm" src="https://user-images.githubusercontent.com/810438/44106669-33f571fc-9fed-11e8-8243-fcb7733089da.png">

* I dislike the "lack" of input background. It's probably not great from accessibility standpoint (very low contrast) and in general I haven't seen this pattern elsewhere. Maybe it would be better if clicking transformed it into an input?

<img width="250" alt="screen shot 2018-08-14 at 6 10 00 pm" src="https://user-images.githubusercontent.com/810438/44106712-4bd67398-9fed-11e8-84df-d4dbc795e53b.png">

bvaughn commented:
The first thing you pointed out was an edge-case when the tail was clicked but then the node collapsed. It's now fixed.

As for the search input style– I like it, but...I don't feel strong enough about it to fight for it 😄so I changed its color when focused like you mentioned.

![kapture 2018-08-14 at 18 28 55](https://user-images.githubusercontent.com/29597/44126989-f2ff47f6-9fef-11e8-8b9b-6a44959ea6e4.gif)

bvaughn commented:
❤️
